src/node_crypto.cc

```diff
@@ -240,6 +240,43 @@ static int PasswordCallback(char *buf, int size, int rwflag, void *u) {
   return 0;
 }
 
+// Loads OpenSSL engine by engine id and returns it. The loaded engine
+// gets a reference so remember the corresponding call to ENGINE_free.
+// In case of error the appropriate js exception is scheduled
+// and nullptr is returned.
+#ifndef OPENSSL_NO_ENGINE
+static ENGINE* LoadEngineById(Environment* env, const char* engine_id) {
+  ENGINE* engine = ENGINE_by_id(engine_id);
```

Comment:
`ENGINE_by_id()` mutates the error stack if the engine cannot be found or loaded.  This line should probably be below the `MarkPopErrorOnReturn` declaration.